packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_navigation_delegate.dart

```diff
@@ -0,0 +1,279 @@
+        return WebResourceErrorType.failedSslHandshake;
+      case android_webview.WebViewClient.errorFile:
+        return WebResourceErrorType.file;
+      case android_webview.WebViewClient.errorFileNotFound:
+        return WebResourceErrorType.fileNotFound;
+      case android_webview.WebViewClient.errorHostLookup:
+        return WebResourceErrorType.hostLookup;
+      case android_webview.WebViewClient.errorIO:
+        return WebResourceErrorType.io;
+      case android_webview.WebViewClient.errorProxyAuthentication:
+        return WebResourceErrorType.proxyAuthentication;
+      case android_webview.WebViewClient.errorRedirectLoop:
+        return WebResourceErrorType.redirectLoop;
+      case android_webview.WebViewClient.errorTimeout:
+        return WebResourceErrorType.timeout;
+      case android_webview.WebViewClient.errorTooManyRequests:
+        return WebResourceErrorType.tooManyRequests;
+      case android_webview.WebViewClient.errorUnknown:
+        return WebResourceErrorType.unknown;
+      case android_webview.WebViewClient.errorUnsafeResource:
+        return WebResourceErrorType.unsafeResource;
+      case android_webview.WebViewClient.errorUnsupportedAuthScheme:
+        return WebResourceErrorType.unsupportedAuthScheme;
+      case android_webview.WebViewClient.errorUnsupportedScheme:
+        return WebResourceErrorType.unsupportedScheme;
+    }
+
+    throw ArgumentError(
+      'Could not find a WebResourceErrorType for errorCode: $errorCode',
+    );
+  }
+}
+
+/// Object specifying creation parameters for creating a [AndroidNavigationDelegate].
+///
+/// When adding additional fields make sure they can be null or have a default
+/// value to avoid breaking changes. See [PlatformNavigationDelegateCreationParams] for
+/// more information.
+@immutable
+class AndroidNavigationDelegateCreationParams
+    extends PlatformNavigationDelegateCreationParams {
+  /// Creates a new [AndroidNavigationDelegateCreationParams] instance.
+  const AndroidNavigationDelegateCreationParams._({
+    this.loadUrl,
+    @visibleForTesting this.androidWebViewProxy = const AndroidWebViewProxy(),
+  }) : super();
+
+  /// Creates a [AndroidNavigationDelegateCreationParams] instance based on [PlatformNavigationDelegateCreationParams].
+  factory AndroidNavigationDelegateCreationParams.fromPlatformNavigationDelegateCreationParams(
+    // Recommended placeholder to prevent being broken by platform interface.
+    // ignore: avoid_unused_constructor_parameters
+    PlatformNavigationDelegateCreationParams params, {
+    LoadUrlCallback? loadUrl,
+    @visibleForTesting
+        AndroidWebViewProxy androidWebViewProxy = const AndroidWebViewProxy(),
+  }) {
+    return AndroidNavigationDelegateCreationParams._(
+      loadUrl: loadUrl,
+      androidWebViewProxy: androidWebViewProxy,
+    );
+  }
+
+  /// Callback responsible for loading the [url] after a navigation request is approved.
+  final LoadUrlCallback? loadUrl;
```

Comment:
Wouldn't this always be null if the class was instantiated by a user? I think this should be a mutable field inside of `AndroidNavigationDelegate`, so that your implementation of `AndroidWebViewController` can set the value in `controller.setNavigationDelegate.`.

Reply:
Not sure I understand, wouldn't the user instantiate the `AndroidNavigationDelegate` using the `AndroidNavigationDelegateCreationParams`, which has a constructor parameter to set the `loadUrl` parameter?

Reply:
The user could do that, but they would most likely use the app-facing `NavigationDelegate` class. This would call `AndroidWebViewPlatform.createPlatformNavigationDelegate`. Therefore, the `loadUrl` would not be set when they call `WebViewController.setNavigationDelegate`.

Reply:
Fixed, converted the creation parameter into a `setOnLoadUrl()` method.